src/problems/MetalChem/metalcloud.cpp

```diff
@@ -0,0 +1,605 @@
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_lo = grid_elem.prob_lo_;
+	amrex::GpuArray<amrex::Real, AMREX_SPACEDIM> prob_hi = grid_elem.prob_hi_;
+	const amrex::Box &indexRange = grid_elem.indexRange_;
+	const amrex::Array4<double> &state_cc = grid_elem.array_;
+
+	std::array<Real, NumSpec> numdens = {-1.0};
+
+	for (int n = 1; n <= NumSpec; ++n) {
+		switch (n) {
+
+			case 1:
+				numdens[n - 1] = userData_.primary_species_1;
+				break;
+			case 2:
+				numdens[n - 1] = userData_.primary_species_2;
+				break;
+			case 3:
+				numdens[n - 1] = userData_.primary_species_3;
+				break;
+			case 4:
+				numdens[n - 1] = userData_.primary_species_4;
+				break;
+			case 5:
+				numdens[n - 1] = userData_.primary_species_5;
+				break;
+			case 6:
+				numdens[n - 1] = userData_.primary_species_6;
+				break;
+			case 7:
+				numdens[n - 1] = userData_.primary_species_7;
+				break;
+			case 8:
+				numdens[n - 1] = userData_.primary_species_8;
+				break;
+			case 9:
+				numdens[n - 1] = userData_.primary_species_9;
+				break;
+			case 10:
+				numdens[n - 1] = userData_.primary_species_10;
+				break;
+			case 11:
+				numdens[n - 1] = userData_.primary_species_11;
+				break;
+			case 12:
+				numdens[n - 1] = userData_.primary_species_12;
+				break;
+			case 13:
+				numdens[n - 1] = userData_.primary_species_13;
+				break;
+			case 14:
+				numdens[n - 1] = userData_.primary_species_14;
+				break;
+			case 15:
+				numdens[n - 1] = userData_.primary_species_15;
+				break;
+			case 16:
+				numdens[n - 1] = userData_.primary_species_16;
+				break;
+			case 17:
+				numdens[n - 1] = userData_.primary_species_17;
+				break;
+			case 18:
+				numdens[n - 1] = userData_.primary_species_18;
+				break;
+			case 19:
+				numdens[n - 1] = userData_.primary_species_19;
+				break;
+			case 20:
+				numdens[n - 1] = userData_.primary_species_20;
+				break;
+			case 21:
+				numdens[n - 1] = userData_.primary_species_21;
+				break;
+			case 22:
+				numdens[n - 1] = userData_.primary_species_22;
+				break;
+			case 23:
+				numdens[n - 1] = userData_.primary_species_23;
+				break;
+			case 24:
+				numdens[n - 1] = userData_.primary_species_24;
+				break;
+			case 25:
+				numdens[n - 1] = userData_.primary_species_25;
+				break;
+			case 26:
+				numdens[n - 1] = userData_.primary_species_26;
+				break;
+			case 27:
+				numdens[n - 1] = userData_.primary_species_27;
+				break;
+			case 28:
+				numdens[n - 1] = userData_.primary_species_28;
+				break;
+			case 29:
+				numdens[n - 1] = userData_.primary_species_29;
+				break;
+			case 30:
+				numdens[n - 1] = userData_.primary_species_30;
+				break;
+			case 31:
+				numdens[n - 1] = userData_.primary_species_31;
+				break;
+			case 32:
+				numdens[n - 1] = userData_.primary_species_32;
+				break;
+			case 33:
+				numdens[n - 1] = userData_.primary_species_33;
+				break;
+			case 34:
+				numdens[n - 1] = userData_.primary_species_34;
+				break;
```

Comment:
Can these be replaced with a macro?

Reply:
yeah, but you still need to call the macro 34 times, no? like the way we did it in microphysics...

Reply:
Yes